pandas/core/indexes/api.py

```diff
@@ -206,6 +206,17 @@ def conv(i):
 
         return Index(lib.fast_unique_multiple_list([conv(i) for i in inds], sort=sort))
 
+    # GH 35092. Detect if we have an Index type for which the sort
+    # setting doesn't make sense
+    ind_types = list({type(index) for index in indexes})
```

Comment:
Is this cast to `list` necessary?

Reply:
@gfyoung 
Sorry, copied it from lower in the code without thinking. An ordinary list comprehension will suffice.